src/QMCWaveFunctions/lcao/CuspCorrectionConstruction.cpp

```diff
@@ -211,55 +223,84 @@ void generateCuspInfo(int orbital_set_size,
   int start_mo = offset[Comm.rank()];
   int end_mo   = offset[Comm.rank() + 1];
   app_log() << "  Number of molecular orbitals to compute correction on this rank: " << end_mo - start_mo << std::endl;
-  for (int mo_idx = start_mo; mo_idx < end_mo; mo_idx++)
+
+#pragma omp parallel
   {
-    app_log() << "   Working on MO: " << mo_idx << std::endl;
+    ParticleSet localTargetPtcl(targetPtcl);
+    ParticleSet localSourcePtcl(sourcePtcl);
+
+    LCAOrbitalSet local_phi(phi);
+    local_phi.myBasisSet = phi.myBasisSet->makeClone();
+    local_phi.IsCloned   = true;
+    local_phi.C          = nullptr;
+    local_phi.setIdentity(false);
+
+    LCAOrbitalSet local_eta(eta);
+    local_eta.myBasisSet = eta.myBasisSet->makeClone();
+    local_eta.IsCloned   = true;
+    local_eta.C          = nullptr;
+    local_eta.setIdentity(false);
+
+// Specify dynamic scheduling explicitly for load balancing.   Each iteration should take enough
+// time that scheduling overhead is not an issue.
+#pragma omp for schedule(dynamic) collapse(2)
     for (int center_idx = 0; center_idx < num_centers; center_idx++)
     {
-      *(eta.C) = *(lcwc.C);
-      *(phi.C) = *(lcwc.C);
-      splitPhiEta(center_idx, corrCenter, phi, eta);
-
-      bool corrO = false;
-      auto& cref(*(phi.C));
-      for (int ip = 0; ip < cref.cols(); ip++)
-      {
-        if (std::abs(cref(mo_idx, ip)) > 0)
-        {
-          corrO = true;
-          break;
-        }
-      }
-
-      if (corrO)
+      for (int mo_idx = start_mo; mo_idx < end_mo; mo_idx++)
       {
-        OneMolecularOrbital etaMO(&targetPtcl, &sourcePtcl, &eta);
-        etaMO.changeOrbital(center_idx, mo_idx);
+        app_log() << "   Working on MO: " << mo_idx << " Center: " << center_idx << std::endl;
```

Comment:
This should be eliminated, the debugger can also tell you every time you are here. Or at least make it debug only. You should not writing from every thread and profiling, since there should be a mutex back there somewhere.

Reply:
I think it's important that there be some feedback to the user that progress is being made.   The loop body is time-consuming enough that any lock here should not be an issue.  I'll add an omp critical pragma to keep the output clean, and to make it clear there is a lock.